Allow superusers to apply waivers

ryscher opened the pull request:
For https://github.com/CDL-Dryad/dryad-product-roadmap/issues/2054

On the Activity log page, if viewing as a superuser:
- Displays the current payment information for this dataset
- Adds a button at the bottom that allows adding a waiver to the dataset, if a waiver is not currently present

sfisher commented on app/views/stash_engine/admin_datasets/curation_activity_error.js.erb:
```diff
@@ -1,6 +1,6 @@
 <% title = @resource.title.html_safe %>
 
-$('#genericModalContent').html("<%= escape_javascript("<h1>#{title}</h1>#{@error_message}") %>");
+$('#genericModalContent').html("<h1><%= title %></h1><%= @error_message %>");
```
You may want to leave the `escape_javascript()` around the user input.  It's not likely but people could add in javascript there that runs.
![Screen Shot 2022-12-05 at 1 45 32 PM](https://user-images.githubusercontent.com/105433/205749392-a95765e3-983a-418e-8a86-3c51b7bfd673.png)

![Screen Shot 2022-12-05 at 1 45 42 PM](https://user-images.githubusercontent.com/105433/205749429-0b992c99-6e66-42e7-8771-30812e0b00f1.png)

![Screen Shot 2022-12-05 at 1 45 51 PM](https://user-images.githubusercontent.com/105433/205749458-a17866d0-d055-4a66-aadb-20ce5b82c6e8.png)

![Screen Shot 2022-12-05 at 1 46 01 PM](https://user-images.githubusercontent.com/105433/205749502-85951c80-0d0a-4c93-a954-f2ddaf9a94fa.png)

sfisher reviewed (approved):
Overall looks good and works well.

ryscher commented:
Ok, updated to reinstate the escaping. I moved it into the individual values, though, since it wasn't rendering properly when it was around the entire HTML.

sfisher commented:
Thanks for updating, Ryan.
